src/json.hpp

```diff
@@ -2702,32 +2644,39 @@ class lexer
     // input management
     /////////////////////
 
-    /// reset yytext
+    /// reset yytext; current character is beginning of token
     void reset() noexcept
     {
-        yylen = 0;
-        start_pos = chars_read - 1;
+        yytext.clear();
+        token_string.clear();
+        token_string.push_back(static_cast<char>(current));
     }
 
     /// get a character from the input
     int get()
     {
         ++chars_read;
-        return next_unget ? (next_unget = false, current)
-               : (current = ia->get_character());
+        int c = current = ia->get_character();
+        token_string.push_back(static_cast<char>(c));
+        return c;
     }
 
-    /// add a character to yytext
-    void add(int c)
+    /// unget current character (return it again on next get)
+    void unget()
     {
-        // resize yytext if necessary; this condition is deemed unlikely,
-        // because we start with a 1024-byte buffer
-        if (JSON_UNLIKELY((yylen + 1 > yytext.capacity())))
+        --chars_read;
+        if (JSON_LIKELY(current != std::char_traits<char>::eof()))
         {
-            yytext.resize(2 * yytext.capacity(), '\0');
+            ia->unget_character();
         }
-        assert(yylen < yytext.size());
-        yytext[yylen++] = static_cast<char>(c);
+        if (! token_string.empty())
+            token_string.pop_back();
+    }
+        
+    /// add a character to yytext
+    void add(int c)
```

Comment:
Same question about `int` vs `char` here.

Reply:
Sure, I'll change that.